spec.html

```diff
@@ -38,6 +38,216 @@ <h1>Well-known intrinsic objects</h1>
+			1. Else,
+				1. Let _wrappedNamespaceValue_ be WrappedModuleNamespaceCreate(_callerRealm_, _namespace_.[[Module]], _namespace_.[[Exports]]).
+				1. Perform ! Call(_promiseCapability_.[[Resolve]], *undefined*, &laquo; _wrappedNamespaceValue_ &raquo;).
+			1. Return ~unused~.
+		1. Let _onFulfilled_ be CreateBuiltinFunction(_fulfilledClosure_, 0, *""*, &laquo; &raquo;).
+		1. Let _rejectedClosure_ be a new Abstract Closure with parameters (_error_) that captures _promiseCapability_ and performs the following steps when called:
+			1. Assert: _error_ is an Error object from _callerRealm_.
+			1. Perform ! Call(_promiseCapability_.[[Reject]], *undefined*, &laquo; _error_ &raquo;).
+			1. Return ~unused~.
+		1. Let _onRejected_ be CreateBuiltinFunction(_rejectedClosure_, 0, *""*, &laquo; &raquo;).
+		1. Perform PerformPromiseThen(_innerPromise_, _onFulfilled_, _onRejected_).
+		1. Return ~unused~.
+	</emu-alg>
+</emu-clause>
+
+<emu-clause id="sec-hostimportvaluemoduledynamically" type="host-defined abstract operation">
+	<h1>
+		HostShadowRealmImportModuleDynamically(
+			_specifier_: a |ModuleSpecifier| String,
+			_promiseCapability_: a PromiseCapability Record,
+			_callerRealm_: a Realm Record,
+			_evalRealm_: a Realm Record,
+			_evalContext_: an execution context,
+		): ~unused~
+	</h1>
+	<dl class="header">
+		<dt>description</dt>
+		<dd>It performs any necessary setup work in order to make available the module corresponding to _specifier_ occurring in the context of _evalContext_ for the _evalRealm_ with no active script or module. It then performs FinishShadowRealmImport to finish the import process.</dd>
+	</dl>
+	<p>An implementation of HostImportValueModuleDynamically must conform to the following requirements:</p>
+
+	<ul>
+		<li>
+		It must return ~unused~. Success or failure must instead be signaled as discussed below.
+		</li>
+		<li>
+		The host environment must conform to one of the two following sets of requirements:
+		<dl>
+			<dt>Success path</dt>
+
+			<dd>
+			<ul>
+				<li>At some future time, the host environment must perform FinishShadowRealmImport(_callerRealm__, _specifier_, _promiseCapability_, _promise_), where _promise_ is a Promise resolved with *undefined*.</li>
+
+				<li>???Any subsequent call to HostResolveImportedModule after FinishShadowRealmImport has completed, given the arguments *null* and _specifier_, must return a normal completion containing a module which has already been evaluated, i.e. whose Evaluate concrete method has already been called and returned a normal completion.</li>
+			</ul>
+			</dd>
+
+			<dt>Failure path</dt>
+
+			<dd>
+			<ul>
+				<li>At some future time, the host environment must perform FinishShadowRealmImport(_callerRealm__, _specifier_, _promiseCapability_, _promise_), where _promise_ is a Promise rejected with an error object from _callerRealm_ representing the cause of failure.</li>
```

Comment:
@syg @legendecas is this enough information for implementers to produce such error from the `_callerRealm_` rather than from the `_evalRealm_`? Or should we be more specific in the spec on how to construct such error?